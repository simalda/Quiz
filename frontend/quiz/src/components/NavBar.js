import React from 'react'
import logo from './LOGO.png';
import { Link } from "react-router-dom";
import "./login.css";

class NavBar extends React.Component {
  constructor(props) {
    super(props);
    this.languages = ["Python", "C#", "JS"];
  }
 


  handlesubmit(submitter, value){
     submitter(value);
}

produceLanguages(languages) {
  return languages.map(language => (
   <Link  key = {language}  to = "/chosenumberofquestions" className="navbar-text" id={language} onClick={() => this.handlesubmit(this.props.onLanguageSelected,language)}>
   {language}  </Link>));
}

  render() {
     
    let addQuest
    if(this.props.user != 'guest' ){
        addQuest = (     <div><Link id="addQuestion"  className="navbar-text " to ="/addquestion">        Add a question        </Link>
        <Link id="stat" onClick={() => this.props.getStat()}  className="navbar-text " to = "/statistics">        Statistics      </Link>
        <div className="navbar-text ">     {this.props.user}        </div> </div>)}
       
    return (
      <nav className="navbar">     
      <div className="navbar-brand">
            <img src={logo} height="40" className="App-logo"  alt=''></img>                 
      </div>
       <div className = "lang">
       <Link id="aboutUs" key = "AboutUs"  className="navbar-text"  to = "/aboutus" onClick={() => this.props.AboutUs()}>
   About Us</Link>
      {this.produceLanguages(this.languages)} </div>
      
         
      {addQuest}
             
      </nav>
    )
  };
}

export default  NavBar 