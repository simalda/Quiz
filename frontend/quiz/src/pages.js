const pages = {
    Login: "Login",//-1
    NumberofQuestions:"NumberofQuestions",//3
    Main: "Main",//1
    QuizStep: "QuizStep",//0
    Results:"Results",//2
    Statistics: "Statistics",//5
    AddQuestion:"AddQuestion",
    AboutUs: "AboutUs"
  };
  
  export default pages