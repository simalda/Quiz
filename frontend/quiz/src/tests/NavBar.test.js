import React from "react";
import ReactDOM from "react-dom";

import Enzyme, { mount, shallow, render } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import pages from "../pages";
import modalKinds from "../ModlKind";
import NavBar from "../components/NavBar";
Enzyme.configure({ adapter: new Adapter() });

 
const state ={
  quiz:[{
    quesion:"a",
    answer:["c","d","s"],
    correctAnswer: "cor"
  },
  {
    quesion:"ab",
    answer:["c","d","s"],
    orrectAnswer: "cor2"
  }],
  highlight:'none',
  page:pages.Login,
  answerOptions:[]
}

describe("NavBar", () => {
  let wrapper;
  let mockSubmit;
  beforeEach(() => {
    mockSubmit = jest.fn();
    wrapper = shallow(<NavBar submit={mockSubmit} />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

   

  it('should show text', ()=>{
    const text = wrapper.find({ id: "aboutUs" });
    expect(text.text()).toBe("About Us");
  })

  it('should find one button for about us', () => {     
    const butto = wrapper.find({id: "aboutUs"})
       expect(butto).toHaveLength(1);
        });  

 
});

