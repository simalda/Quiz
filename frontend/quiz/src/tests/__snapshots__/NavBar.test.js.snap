// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`NavBar should match the snapshot 1`] = `ShallowWrapper {}`;
